File: app/src/main/java/com/zacharee1/systemuituner/FragmentHelpers/TWHelper.java
package com.zacharee1.systemuituner.FragmentHelpers;

import android.preference.Preference;
import android.preference.SwitchPreference;
import android.provider.Settings;

import com.zacharee1.systemuituner.ItemDetailFragment;
import com.zacharee1.systemuituner.Utils.SettingsUtils;

/**
 * Created by Zacha on 7/16/2017.
 */

public class TWHelper
{
    private ItemDetailFragment mFragment;

    public TWHelper(ItemDetailFragment fragment) {
        mFragment = fragment;

        setHBWState();
        setHBWListener();
    }

    private void setHBWState() {
        String hbw = Settings.Global.getString(mFragment.getContext().getContentResolver(), "limit_brightness_state");
        SwitchPreference hbwSwitch = (SwitchPreference) mFragment.findPreference("high_bright_warning");

        if (hbw == null || hbw.isEmpty()) {
            hbwSwitch.setChecked(false);
        } else {
            hbwSwitch.setChecked(true);
        }
    }

    private void setHBWListener() {
        SwitchPreference hbwSwitch = (SwitchPreference) mFragment.findPreference("high_bright_warning");

        hbwSwitch.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener()
        {
            @Override
            public boolean onPreferenceChange(Preference preference, Object o)
            {
                if (Boolean.valueOf(o.toString())) {
                    SettingsUtils.writeGlobal(mFragment.getContext(), "limit_brightness_state", "80,80");
                } else {
                    SettingsUtils.writeGlobal(mFragment.getContext(), "limit_brightness_state", "");
                }
                return true;
            }
        });
    }
}

File: app/src/main/java/com/zacharee1/systemuituner/fragmenthelpers/TWHelper.kt
package com.zacharee1.systemuituner.fragmenthelpers

import android.graphics.Color
import android.preference.Preference
import android.provider.Settings

import com.jaredrummler.android.colorpicker.ColorPreference
import com.zacharee1.sliderpreferenceembedded.SliderPreferenceEmbedded
import com.zacharee1.systemuituner.fragments.ItemDetailFragment
import com.zacharee1.systemuituner.util.SettingsUtils

class TWHelper(fragment: ItemDetailFragment) : BaseHelper(fragment) {

    private val statbarHelper: StatbarHelper = StatbarHelper(fragment)

    init {
        setUpQSStuff()
        setUpNavBarStuff()
    }

    private fun setUpQSStuff() {
        val rows = findPreference(TILE_ROW) as SliderPreferenceEmbedded
        val columns = findPreference(TILE_COLUMN) as SliderPreferenceEmbedded
        val defVal = 3
        val savedRowVal = Settings.Secure.getInt(activity?.contentResolver, rows.key, defVal)
        val savedColVal = Settings.Secure.getInt(activity?.contentResolver, columns.key, defVal)

        val listener = Preference.OnPreferenceChangeListener { preference, newValue ->
            sharedPreferences.edit().putInt(preference.key, Integer.valueOf(newValue.toString())).apply()
            SettingsUtils.writeSecure(activity, preference.key, newValue.toString())
            true
        }

        rows.onPreferenceChangeListener = listener
        columns.onPreferenceChangeListener = listener

        rows.progress = savedRowVal
        columns.progress = savedColVal
    }

    private fun setUpNavBarStuff() {
        val preference = findPreference(NAVBAR_COLOR) as ColorPreference
        val savedVal = Settings.Global.getInt(context?.contentResolver, NAVBAR_COLOR, Color.WHITE)

        preference.saveValue(savedVal)
        preference.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _, newValue ->
            SettingsUtils.writeGlobal(context, NAVBAR_COLOR, newValue.toString())
            SettingsUtils.writeGlobal(context, NAVBAR_CURRENT_COLOR, newValue.toString())
            true
        }
    }

    override fun onDestroy() {
        statbarHelper.onDestroy()
    }

    companion object {
        const val TILE_ROW = "qs_tile_row"
        const val TILE_COLUMN = "qs_tile_column"
        const val NAVBAR_COLOR = "navigationbar_color"
        const val NAVBAR_CURRENT_COLOR = "navigationbar_current_color"
    }
}
